### PR TITLE
lint: WIP markdown support

### PR DESCRIPTION
**Pull request by woodruffw**
See #7.

Signed-off-by: William Woodruff <william@trailofbits.com>

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -52,11 +52,20 @@ env:
   # the subset of development dependencies installed.
   INSTALL_EXTRA: lint
 
+  ORG_REPO: trailofbits.com/.github
+  ORG_REPO_REF: ${{ github.job_workflow_sha }}
+  ORG_REPO_PATH: /tmp/org_github
```
Comment (woodruffw) [inline .github/workflows/lint.yml]: Wait, does this work? 🤯 

For some reason I had trouble getting the `job_workflow_sha` to evaluate to anything nonempty, but maybe I was using it in the wrong context.

Comment (oldsj) [reply .github/workflows/lint.yml]: not sure yet, its still failing on my env syntax on ORG_REPO 😄

Comment (oldsj) [reply .github/workflows/lint.yml]: looks like no, this doesn't work. I think it'll be fine to just use main

Comment (woodruffw) [reply .github/workflows/lint.yml]: Yeah, that works for me (I wanted to avoid it because it isn't pinned, but IMO that's relatively minor since we don't expect to change this file much.)

Comment (oldsj) [reply .github/workflows/lint.yml]: d'oh I had the org repo wrong. I tried it again after fixing that and it does work, but its the head ref of main for some reason 🤔

Comment (oldsj) [reply .github/workflows/lint.yml]: I just made it a configurable input so at least we have that. Other than that this seems to be working 🎉

Comment (woodruffw) [reply .github/workflows/lint.yml]: Nice!

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -52,11 +57,21 @@ env:
...
   lint:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{env.ORG_REPO}}
+          ref: ${{env.ORG_REPO_REF}}
+          path: tmp/org_repo
+      - run: mv tmp/org_repo ${ORG_REPO_PATH}
```
Comment (woodruffw) [inline .github/workflows/lint.yml]: Out of curiosity: why is this necessary? Can't we just clone directly to `ORG_REPO_PATH`?

**Comment by woodruffw** [comment]
NB: I dropped `MD043` in [cfecdac](https://github.com/trailofbits/.github/pull/12/commits/cfecdac3ae4139a126fbba5e2f3a4260bc48532e) because it was creating lint results that didn't make sense to me (AFAICT it had the default config, but was producing errors on reasonable looking headers).

File: .github/workflows/lint.yml (hunk 2 above)
Comment (woodruffw) [reply .github/workflows/lint.yml]: TIL `actions/checkout` doesn't allow the `path:` to be outside of the runner's directory!
